### ISSUE TITLE
Make it easier to reach rooms belonging to a space

### ISSUE THREAD

**Issue by frandavid100** [vector-im/element-android#3078]
#### Is your feature request related to a problem? Please describe.

Users coming from other chat applications (Like Whatsapp or Telegram) are used to have all their conversations in plain view. Element has traditionally made it more complex, by splitting conversations into two different tabs (DMs and Rooms). Because of this, I have already received complaints from casual users who weren't able to find a conversation they had with me.

However, the addition of spaces brings a new layer of complexity. If I share a room with a family member, and that room belongs to a space, that person needs to take the following steps in order to find the room:

1. Click the Home button at the top-left corner, which is already unintuitive because there's no clue that it's even clickable.
2. Scroll, find and click the corresponding space.
3. Click the Rooms tab.
4. Find and click the room.

I'm worried that this will be too cumbersome and unintuitive for casual users, and I'd like to propose an alternative solution:

#### Describe the solution you'd like.

Instead of a navigation drawer, I would use a filter bar similar to the one used by other apps like Google Play Books. Here's a mockup:

![image](https://user-images.githubusercontent.com/694794/112745620-1b4a5280-8faa-11eb-828d-62d4119a91f8.png)

In my mockup spaces are always shown on the filter bar at the top, giving the user a clear graphical clue that they're there. Home is selected by default, but any other space can be clicked at any time in order to change the set of rooms that are shown.

#### What about subspaces?

My design also takes care of them. When the user selects a space, all subspaces belonging to it appear to the right and can also be clicked. Here's a mockup:

![image](https://user-images.githubusercontent.com/694794/112745761-0a4e1100-8fab-11eb-8c23-5a0642631b8a.png)

The same applies to sub-subspaces. When the user selects a space, al sub-subspaces belonging to it appear to the right:

![image](https://user-images.githubusercontent.com/694794/112745843-a11acd80-8fab-11eb-8edc-46bd5ecddea5.png)

This can go as deep as needed, regardless of how many levels the user's tree has. Which in my opinion also makes it more suitable than the navigation drawer for some power users.

In my opinion, the main advantages of this design are that:

1. Spaces are always in plain sight, as opposed to hidden in a navigation drawer. This is important when the user wants to find a room, but also when a space has notifications that require attention.
2. Spaces can be reached more easily, with just one click as opposed to two.
3. Subspaces are also easier to find and reach, because of the same reason.
4. It might be difficult to show deeply nested spaces in the navigation drawer, but there's no such problem here.

**Comment by timokoesters** [comment]
Note that spaces are entirely optional. If you think a user will be confused by spaces you can just not invite them.

**Comment by frandavid100** [comment]
> Note that spaces are entirely optional. If you think a user will be confused by spaces you can just not invite them.

Indeed. But I still think that making the feature easy to use for them is a better option.

**Comment by kittykat** [comment]
@frandavid100 Now that Spaces has been released, if there are any changes to your original issue, can you please update it?
